plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -127,6 +127,8 @@ fu_mediatek_scaler_device_use_aux_dev(FuDevice *device, GError **error)
 
 		udev = fu_udev_device_new(fu_device_get_context(device), element->data);
 		i2c_devices = fu_udev_device_get_siblings_with_subsystem(udev, "i2c");
+		if (i2c_devices->len == 0)
+			continue;
```

Comment:
I thought it eventually report the same error as current flow is?

Reply:
I think what coverity is saying is "you're always choosing the first device, why bother with the loop"

Reply:
So do you think my first patch is appropriate, or should we do something like the fix-up instead?

Reply:
Yes, you are correct. We actually don't need iterating the dp_aux devices again. I've sent another patch if that change makes sensible to you.

Reply:
Better indeed! Thanks.